payments-core/src/main/java/com/stripe/android/StripePaymentController.kt

```diff
@@ -111,6 +111,12 @@ internal class StripePaymentController internal constructor(
         )
     }
 
+    /**
+     * returnUrl passed to [ConfirmPaymentIntentParams] or [ConfirmSetupIntentParams], this value
+     * will change upon any new intent confirmation.
+     */
+    private var returnUrl: String? = null
```

Comment:
This feels like it's making the code more brittle. Is there any chance for a race condition?

If we're going this route, I think it would make sense to at least have a map of intent id to return url.

Reply:
Replaced with a map, the entry will be consumed when 3ds1 typed intent is confirmed.
Since the authenticator query happens after we override the returnUrl, it won't cause race condition inside. (If a client has one `Stripe` instance and call that from different threads, it's possible this would break. But I guess that's not something we expect user to do anyway and there might other stateful properties that are not threadsafe within the SDK atm)